sdk/cosmos/azure_data_cosmos/src/routing/collection_cache.rs

```diff
@@ -0,0 +1,271 @@
+// cSpell:ignore smol
+#![allow(dead_code)]
+
+use super::async_cache::AsyncCache;
+use crate::cosmos_request::CosmosRequest;
+use crate::handler::retry_handler::{BackOffRetryHandler, RetryHandler};
+use crate::operation_context::OperationType;
+use crate::routing::global_endpoint_manager::GlobalEndpointManager;
+use crate::{models::ContainerProperties, resource_context::ResourceLink, ReadContainerOptions};
+use azure_core::http::{Pipeline, Response};
+use azure_core::Error;
+use std::time::Duration;
+
+/// Cache for Cosmos DB container metadata and properties.
+///
+/// # Summary
+/// Maintains an in-memory cache of container properties (partition keys, indexing policies, etc.)
+/// to minimize redundant metadata requests to the Cosmos DB service. Uses a 5-minute TTL by default
+/// to balance freshness with performance. Integrates with retry handler for resilient metadata fetching
+/// across regional endpoints.
+#[derive(Clone, Debug)]
+pub struct CollectionCache {
+    pipeline: Pipeline,
+    global_endpoint_manager: GlobalEndpointManager,
+    container_properties_cache: AsyncCache<String, ContainerProperties>,
+    retry_handler: BackOffRetryHandler,
+}
+
+impl CollectionCache {
```

Comment:
nit: Let's use `ContainerCache` as the name here. Collection is a legacy name, and I understand why we have code in older SDKs that references it, but let's try to keep to the newer terms here (Item, Container, etc.)